### PR TITLE
docs: rework the Architecture section

### LINKED ISSUE #239: Architecture v2
Re work the architecture page, it should be updated to provide a better overview

### PR DESCRIPTION
**Pull request by fgiudici**
Related to #239

### CODE REVIEW

**Review by kkaempf — APPROVED** [review]
Merge it.
We'll improve it iteratively over the coming weeks.

File: docs/architecture.md [hunk 1]
```diff
@@ -1,42 +1,104 @@
...
+- MachineRegistration
+- MachineInventory
+- MachineInventorySelector
+- MachineInventorySelectorTemplate
+- ManagedOSImage
+- ManagedOSVersion
+- ManagedOSVersionChannel
+
+#### MachineRegistration
+A MachineRegistration resource should be created for each new host to be provisioned by Elemental.
```
Comment (alexander-demichev) [inline docs/architecture.md:47]: My impression was that machine registration can manage a set of hosts

File: docs/architecture.md [hunk 2]
```diff
@@ -1,42 +1,104 @@
...
+
+The MachineRegistration has a `Ready` condition which turns to true when the {{elemental.operator.name}} has successfully generated the registration URL and an associated ServiceAccount. From this point on the target host can connect to the registration URL to kick off the provisioning process.
+
+An HTTP GET request against the registration URL returns the _registration file_: a .yaml file containing the registration data (i.e., the _spec:config:elemental:registration_ section only from the just created MachineRegistration).
+The registration file contains all the required data to allow the target host to perform self registration and start the Elemental provisioning. See the [config:elemental:registration section in the MachineRegistration reference](machineregistration-reference#configelementalregistration)) for more details on the available registration options.
+
+
+#### MachineInventory
+When a new host registers successfully, the {{elemental.operator.name}} creates a MachineInventory resource representing that particular host.
```
Comment (alexander-demichev) [inline docs/architecture.md:59]: We should note that it's created by host agent and not by operator running on the management cluster

File: docs/architecture.md [hunk 3]
```diff
@@ -1,42 +1,104 @@
...
+The MachineInventory stores the TPM hash of the tracked host, retrieved during the registration process, and allows to execute arbitrary commands (plans) on the machine.
+
+A MachineInventory has three conditions:
+
+- `Initialized`, tracking if the resources needed for applying the plan have been correctly created.
+- `PlanReady`, showing if the host has completed its current plan.
+- `Ready`, which indicates that a machine has been initialized and has no running plans.
+
+#### MachineInventorySelector
+A MachineInventorySelector selects MachineInventories based on applied selectors (usually patter matching on MachineInventory label values).
```
Comment (alexander-demichev) [inline docs/architecture.md:69]: should we note that it implements a CAPI infrastructure provider?

File: docs/architecture.md [hunk 4]
```diff
@@ -1,42 +1,104 @@
...
+
+MachineInventorySelectors have three conditions:
+
+- `InventoryReady`, turn to true if the MachineInventorySelector has found a matching MachineInventory and has successful set itself as the MachineInventorySelector owner.
+- `BootstrapReady`, reports if the selector has successfully applied its bootstrap plan.
+- `Ready`, tracks if the inventory has been correctly selected and bootstrapped.
+
+#### MachineInventorySelectorTemplate
+The MachineInventorySelectorTemplate is a user defined resource that will be used as the blueprint to create the required MachineInventorySelectors: it includes the selector to identify the eligible MachineInventories.
```
Comment (alexander-demichev) [inline docs/architecture.md:78]: same here, should we note that it's a CAPI infrastructure template?

File: docs/architecture.md [hunk 5]
```diff
@@ -1,42 +1,104 @@
...
+
+### {{elemental.register.name}}
+New hosts start the Elemental provisioning process through the {{elemental.register.name}}: this tool requires a valid elemental-operator registration URL as input, and performs the following steps:
+
+- setups a websocket connection to the registration URL
+- authenticates itself using the registration token and the onboard TPM<sup>1</sup>
+- sends [SMBIOS data](smbios.md) to the {{elemental.operator.name}}
+- retrieves the Elemental installation configuration
+- starts the {{elemental.cli.name}} and performs the Elemental OS installation
+- reboots in the newly installed Elemental OS
```
Comment (alexander-demichev) [inline docs/architecture.md:89]: thanks for documenting this!

File: docs/architecture.md (hunk 1 above)
Comment (fgiudici) [reply docs/architecture.md:47]: yep right, dropping this liine

**Review by davidcassany — APPROVED** [review]
Nice, I like it. Thanks much to take the time the document the purpose of the elemental operator CRDs.

**Comment by fgiudici** [comment]
Starting merging this as a first step, comments from @alexander-demichev and more coverage to be added in following PRs.

**Pull request merged**